packages/protvista-structure/src/position-mapping.ts

```diff
@@ -0,0 +1,87 @@
+export type Direction = "UP_PDB" | "PDB_UP";
+
+/* eslint-disable camelcase */
+export type Mappings = Array<{
+  entity_id: number;
+  chain_id: string;
+  unp_end: number; // UniProt end coordinate
+  unp_start: number; // UniProt start coordinate
+  struct_asym_id: string;
+  start: {
+    residue_number: number; // PDB start coordinate
+    author_insertion_code: string;
+    author_residue_number: number;
+  };
+  end: {
+    residue_number: number; // PDB end coordinate
+    author_insertion_code: string;
+    author_residue_number: number;
+  };
+}>;
+/* eslint-enable camelcase */
+
+export class PositionMappingError extends Error {}
+
+/**
+ * Translate between UniProt and PDBe positions using SIFTs mappings
+ * @function translatePositions
+ * @param  {Number}     start            The start index for the sequence (1-based)
+ * @param  {Number}     end              The end index for the sequence (1-based)
+ * @param  {Mappings}   mappings         The array of mapping objects
+ * @param  {String}     mappingDirection Indicates direction of maping: UniProt to PDB or PDB to UniProt
+ * @return {Translated}                  Object with: mapped entity ID; mapped chain ID; translated start & end positions
+ */
+const translatePositions = (
+  start: number,
+  end: number,
+  mappings: Mappings,
+  mappingDirection: Direction = "UP_PDB"
+): {
+  start: number;
+  end: number;
+  entity: number;
+  chain: string;
+} | null => {
+  // return if they have been set to 'undefined'
+  if (!start || !end || Number.isNaN(start) || Number.isNaN(end)) {
```

Comment:
There is an additional check in the [original file](https://github.com/ebi-webcomponents/nightingale/blob/master/packages/protvista-structure/src/protvista-structure.js#L304):
```
if (
  typeof this.highlight === "string" ||  <--- do we need this in translatePositions or check it before calling translatePositions
  Number.isNaN(start) ||
  Number.isNaN(end)
) {
  return null;
}
```

Reply:
Checking that it's the correct type could be done here https://github.com/ebi-webcomponents/nightingale/pull/154/files#diff-ec9fd2d846b88c27b3683134bfbc0eea81436cf17c477b8b68dc97fd8ef273fcR146 or in the `attributeChangedCallback`

Reply:
Added a check in the `highlightChain` before `translatePositions` is called.